chunks.go

```diff
@@ -54,6 +54,46 @@ func (cm *ChunkMeta) writeHash(h hash.Hash) error {
 	return nil
 }
 
+// deletedIterator wraps an Iterator and makes sure any deleted metrics are not
+// returned.
+type deletedIterator struct {
+	it chunks.Iterator
+
+	intervals intervals
+}
```

Comment:
Is there a reason for applying the tombstones to each chunk iterator rather than wrapping a full series iterator? Dragging it this far down makes it more invasive.